storage.bs

```diff
@@ -94,37 +100,252 @@ function retrieveNextChunk(nextChunkInfo) {
- <dt>Storage
+ <dt id=site-storage>Storage
  <dd>Indexed DB, Cache API, service worker registrations, <code>localStorage</code>,
  <code>history.pushState()</code>, application caches, notifications, etc.
 </dl>
 
-<p>This standard primarily concerns itself with <dfn export id=site-storage>storage</dfn>.
+<p>This standard primarily concerns itself with storage.
+
+
+
+<h2 id=model>Model</h2>
+
+
+<h3 id=storage-units>Storage units</h3>
+
+<p>A <dfn>storage type</dfn> is "<code>more-durable</code>" or "<code>session</code>".
+
+<p>A <dfn>storage key</dfn> is an <a for=/>origin</a>. [[HTML]]
+
+<p class=XXX>This is expected to change, see
+<a href="https://privacycg.github.io/storage-partitioning/">Client-Side Storage Partitioning</a>.
+
+<p>A <dfn id=site-storage-unit>storage unit</dfn> is the least granular unit of storage, other than
+the user agent. It holds a <dfn>bucket map</dfn>, which is a <a for=/>map</a> of
+<a for=/>strings</a> to <a>storage buckets</a>.
+
+<p class="note">For now "<code>default</code>" is the only <a for=map>key</a> that exists. See
+<a href="https://github.com/whatwg/storage/issues/2">issue #2</a>. It is given a
+<a for=map>value</a> when a <a>storage unit</a> is <a lt="obtain a storage unit">obtained</a> for
+the first time.
+
+<p>A <dfn>storage map</dfn> is a <a for=/>map</a> of <a>storage keys</a> to <a>storage units</a>. It
+is initially empty.
+
+<p>To <dfn>obtain a storage unit</dfn>, given a <a for=/>storage map</a> <var>map</var>, an
+<a>environment settings object</a> <var>environment</var>, and a <a>storage type</a>
+<var>type</var>, run these steps:
+
+<ol>
+ <li><p>Let <var>key</var> be <var>environment</var>'s
+ <a for="environment settings object">origin</a>.
+
+ <li><p>If <var>key</var> is an <a>opaque origin</a>, then return failure.
+
+ <li><p>If the user has disabled storage, then return failure.
+
+ <li>
+  <p>If <var>map</var>[<var>key</var>] does not <a for=map>exist</a>, then:
+
+  <ol>
+   <li><p>Let <var>unit</var> be a new <a>storage unit</a>.
+
+   <li><p>Set <var>unit</var>'s <a>bucket map</a>["<code>default</code>"] to the result of
+   <a>create a storage bucket</a> with <var>type</var>.
+
+   <li><p>Set <var>map</var>[<var>key</var>] to <var>unit</var>.
+  </ol>
+
+ <li><p>Return <var>map</var>[<var>key</var>].
+</ol>
+
+<hr>
+
+<p>A <a for=/>user agent</a> holds a <dfn for="user agent">storage map</dfn>, which is a
+<a for=/>storage map</a>.
+
+<p>A browsing session holds a <dfn for="browsing session">storage map</dfn>, which is a
+<a for=/>storage map</a>.
+
+<p class=XXX>Browsing session is yet to be formally defined. For all intents and purposes it is a
+<a>top-level browsing context</a>, except that it survives the <a>top-level browsing context</a>
+being replaced due to a cross-origin opener policy.
+
+
+<h3 id=storage-endpoints>Storage endpoints</h3>
+
+<p>A <dfn export>storage endpoint</dfn> is a more-durable or session storage API that uses the
```

Comment:
Do you want to quote/style "more-durable" and "session" here, for consistency?

Reply:
Here they are used in an informal manner to describe the APIs. Elsewhere they're a type value.